src/components/MultiHopTrade/components/MultiHopTradeConfirm/components/FirstHop.tsx

```diff
@@ -95,13 +187,49 @@ export const FirstHop = ({
     }
   }, [tradeExecutionStatus, isApprovalNeeded])
 
+  const { txLink, txHash } = useMemo(() => {
+    if (buyTxHash)
+      return {
+        txLink: getTxLink({
+          name: tradeQuoteStep?.source,
+          defaultExplorerBaseUrl: tradeQuoteStep?.sellAsset.explorerTxLink ?? '',
+          tradeId: buyTxHash,
+        }),
+        txHash: buyTxHash,
+      }
+    if (sellTxHash)
+      return {
+        txLink: getTxLink({
+          name: tradeQuoteStep?.source,
+          defaultExplorerBaseUrl: tradeQuoteStep?.sellAsset.explorerTxLink ?? '',
+          tradeId: sellTxHash,
+        }),
+        txHash: buyTxHash,
+      }
+
+    return {}
+  }, [buyTxHash, sellTxHash, tradeQuoteStep?.sellAsset.explorerTxLink, tradeQuoteStep?.source])
+
+  const {
+    buyAsset,
+    sellAsset,
+    sellAmountIncludingProtocolFeesCryptoBaseUnit,
+    buyAmountBeforeFeesCryptoBaseUnit,
+  } = tradeQuoteStep
+
+  // the txStatus needs to be undefined before the tx is executed to handle "ready" but not "executing" status
+  const approvalTxStatus =
+    tradeExecutionStatus >= MultiHopExecutionStatus.Hop1AwaitingApprovalExecution
```

Comment:
Tangential but wondering if `MultiHopExecutionStatus` could be an enum with only keys to ensure values are numeric and we don't get the ordering wrong, since IIRC, it compiles to a an object where the values are numbers

Reply:
This was an intentional decision to ensure we are deliberate about the ordering of the entries, however I'm going to refactor the enum of be strings and create a function to decide >, <, incrementing etc so we are way less likely to mess things up (i've already done it several times)